public/js/gf-giftaid-field-frontend.js

```diff
@@ -1,57 +1,91 @@
-function gfGiftAidOnInputChange(elem) {
-  if (!(elem instanceof HTMLElement)) {
-    return;
+/**
+ * The donation total object.
+ * @typedef {Object} Donation
+ * @property {string} donation - The amount donated as a string.
+ * @property {string} giftAidTotal - The amount including taxback as a string.
+ */
+
+document.addEventListener('DOMContentLoaded', () => {
+  initGiftAid();
+});
+
+/**
+ * Updates the gift aid display, based on a chosen field changing.
+ */
+function initGiftAid() {
+  const totalSelector = totalFieldSelector('.ginput_amount');
+  window.gform.addAction('gform_input_change', (elem) => {
+    const total = getTotalAmount(elem, totalSelector);
+    if (! total || !total.donation || !total.giftAidTotal) {
+      return;
+    }
+    updateGiftAidDisplay(total);
+  }, 10);
+}
+
+/**
+ * Gets the selector of the total field, selected in the gift aid field settings.
+ * Will be in the format 'field_<form_id>_<field_id>' and will be an id so the selector will be prefixed by #.
+ * @param {string} fallback fallback selector if the field is not found.
+ * @returns {string} The class of the total field.
+ */
+function totalFieldSelector(fallback) {
+  const giftAidComponent = document.querySelector('.gift-box-wrapper');
+  if (! (giftAidComponent instanceof HTMLElement)) {
+    return fallback;
+  }
+  const selectedPriceFieldId = giftAidComponent.dataset.selectedPriceFieldId;
+  if (!selectedPriceFieldId) {
+    return fallback;
   }
 
-  if (! elem.classList.contains('ginput_total') && ! elem.classList.contains('ginput_amount')) {
-    return;
+  return `#${selectedPriceFieldId}`;
+}
+
+/**
+ * Gets the total amount donated and updates the Gift Aid total.
+ * @param {HTMLElement} elem The element that has changed.
+ * @param {string} totalSelector selector of the chosen total field.
+ * @returns {Donation} The total amount donated.
+ */
+function getTotalAmount(elem, totalSelector) {
+  if (! (elem instanceof HTMLInputElement) || ! elem.closest(totalSelector)) {
+    return '';
   }
+  const totalValueStr = elem.value;
+  const totalValueFloat = parseFloat(totalValueStr);
+  if (! totalValueFloat) {
+    return '';
+  }
+  const giftAidAmount = totalValueFloat * 1.25;
+
+  return {
+    donation: totalValueFloat.toFixed(2),
+    giftAidTotal: giftAidAmount.toFixed(2)
+  };
+}
 
-  const parentForm = elem.closest('form');
-  if (!(parentForm instanceof HTMLFormElement)) {
+/**
+ * Updates the gift aid display.
+ * @param {Donation} total
+ * @returns {void}
+ */
+function updateGiftAidDisplay(total) {
+  if (!total || !total.donation || !total.giftAidTotal) {
     return;
   }
-
-  const totalEl = parentForm.querySelector('.ginput_total');
-  if (totalEl instanceof HTMLElement) {
-    elem = totalEl;
+  const gravityForm = document.querySelector('.gform_wrapper');
```

Comment:
do not query entire document.
accept additional argument for form id or form element or something.